deeppath/deeppath.py

```diff
@@ -32,51 +33,78 @@ def flatten(nested_iterable: Iterable[Any]) -> List[Any]:
     return flattened_list
 
 
-def _get_repetition_index(key: str) -> Optional[Tuple[str, int]]:
+def _get_repetition_index(key: str) -> Optional[Tuple[str, Union[int, str]]]:
     """Try to match a path for a repetition.
     This will return the key and the repetition index or None if the key
     does not match the expected regex"""
     match = _REPETITION_REGEX.search(key)
     if match:
         key = match.group(1)
         repetition_number = match.group(2)
-        return key, int(repetition_number)
+        if repetition_number != "*":
+            return key, int(repetition_number)
+        return key, repetition_number
 
     return None
 
 
-def _flatdget(data: Union[Sequence[Any], Mapping, str], key: Union[int, str]) -> Any:
-    if isinstance(data, Sequence) and isinstance(key, int):
-        return data[key]
-    if isinstance(data, Mapping) and isinstance(key, str):
-        return data[key]
-    return [_flatdget(value, key) for value in data]
-
-
-def dget(input_dict: Mapping, path: str, default: Any = None) -> Any:
-    """Gets a deeply nested value in a mapping.
-    Returns default if provided when any key doesn't match.
-    """
-    path = path.strip("/")
-    data: Union[List[Any], Mapping] = input_dict
-    try:
-        for key in path.split("/"):
-            repetition = _get_repetition_index(key)
-            if repetition:
-                key, index = repetition
-                if key != "*":
-                    data = _flatdget(data, key)
-                    data = _flatdget(data, index)
-                elif isinstance(data, Dict):
-                    data = [_flatdget(value, index) for value in data.values()]
-            else:
-                if key != "*":
-                    data = _flatdget(data, key)
-                elif not isinstance(data, Sequence):
-                    data = list(data.values())
-    except (KeyError, TypeError, IndexError):
+def _get_sequence_index(path: str) -> Tuple[bool, Union[None, int, str]]:
+    match = re.match(r"\[(-?[\d\*]+)\]", path)
+    if match:
+        index = match.group(1)
+        if index.isnumeric() or index[0] == "-" and index[1:].isnumeric():
+            return True, int(index)
+        else:
+            return True, index
+    if path == "*":
+        return False, "*"
+    return False, None
+
+
+def dget(
+    data: Dict[str, Any], path: str, default: Optional[Any] = None
```

Comment:
```suggestion
    data: Mapping, path: str, default: Optional[Any] = None
```
Type hint changed from before, is this intentional?